File: Kotlin/src/K09.kt
import java.util.*

class K09 : IExecutable() {
    override fun execute(num: Int) {
        when (num) {
            1 -> question_01()
            2 -> question_02()
            3 -> question_03()
            4 -> question_04()
            5 -> question_05()
        }
    }

    override fun question_01() {
        val ages = IntArray(3)
        val reader = Scanner(System.`in`)
        for (i in 0..2) {
            print("${i + 1}人目の年齢を入力＞")
            ages[i] = reader.nextInt()
        }

        for (i in 1..20) {
            print("-")
        }
        println()

        for ((index, value) in ages.withIndex()) {
            println("${index + 1}人目：${value}歳。")
        }

        println("平均年齢：%.1f歳。".format(ages.average()))
    }

    override fun question_02() {
        val numbers: IntArray = intArrayOf(8, 3, 12, 7, 9)
        print("元々の配列：")
        for (num in numbers) {
            print("$num ")
        }
        println()
        print("逆順での表示：")
        for (num in numbers.reversedArray()) {
            print("$num ")
        }
    }

    override fun question_03() {
        val studentScores = arrayOf(arrayOf(52, 71, 61, 47), arrayOf(6, 84, 81, 20), arrayOf(73, 98, 94, 95))
        println("\t\t|\t科目A\t科目B\t科目C\t科目D")

        for (i in 1..45) {
            print("-")
        }
        println()

        for ((index, student) in studentScores.withIndex()) {
            print("学生${index + 1}\t|\t")
            for (score in student) {
                print("$score\t\t")
            }
            println()
        }
    }

    override fun question_04() {
        val studentScores = arrayOf(arrayOf(52, 71, 61, 47), arrayOf(6, 84, 81, 20), arrayOf(73, 98, 94, 95))
        println("\t\t|\t科目A\t科目B\t科目C\t科目D\t|\t合計点")
        var total = 0
        val average = FloatArray(4)

        for (i in 1..65) {
            print("-")
        }
        println()

        for ((index, student) in studentScores.withIndex()) {
            print("学生${index + 1}\t|\t")
            for ((index2, score) in student.withIndex()) {
                print("$score\t\t")
                total += score
                average[index2] = average[index2] + score
            }
            print("|\t$total")
            total = 0
            println()
        }
        print("平均点\t|\t")
        for (num in average) {
            print("%.1f\t".format(num / 3.0))
        }
        println("|")
    }

    fun question_05() {
        var count = 0
        var numbers = IntArray(100)
        val reader = Scanner(System.`in`)
        for (i in 0 until numbers.size) {
            print("${i + 1}件目の入力：")
            numbers[i] = reader.nextInt()
            if (numbers[i] < 0)
                break
            count++
        }

        println("----並び替え後----")

        for (num in numbers.take(count).sorted()) {
            println(num)
        }
    }
}

File: CSharp/k09.cs
using System;
using System.Linq;

namespace CSharp
{
    class K09 : IExecutable
    {
        public override void Execute(int num)
        {
            switch (num)
            {
                case 1:
                    Question01();
                    break;
                case 2:
                    Question02();
                    break;
                case 3:
                    Question03();
                    break;
                case 4:
                    Question04();
                    break;
                case 5:
                    Question05();
                    break;
            }
        }

        public override void Question01()
        {
            try
            {
                int[] ages = new int[3];
                for (int i = 0; i < 3; i++)
                {
                    Console.Write($"{i + 1}人目の年齢を入力＞");
                    ages[i] = int.Parse(Console.ReadLine());
                }

                for (int i = 0; i < 21; i++)
                {
                    Console.Write("-");
                }
                Console.WriteLine();

                for (int i = 0; i < 3; i++)
                {
                    Console.WriteLine($"{i + 1}人目：{ages[i]}歳。");
                }

                Console.Write($"平均年齢：{Math.Round(ages.Average(), 1)}歳。");
            }
            catch (Exception e)
            {
                Console.Error.WriteLine(e.Message);
            }

        }

        public override void Question02()
        {
            int[] numbers = {8, 3, 12, 7, 9};
            Console.Write("元々の配列：");
            foreach (var num in numbers)
            {
                Console.Write($"{num} ");
            }
            Console.WriteLine();

            Console.Write("逆順での表示：");

            foreach (var num in numbers.Reverse())
            {
                Console.Write($"{num} ");
            }

        }

        public override void Question03()
        {
            int[,] studentScores =
            {
                { 52, 71, 61, 47 },
                { 6, 84, 81, 20 },
                { 73, 98, 94, 95 }
            };

            Console.WriteLine("\t|\t科目A\t科目B\t科目C\t科目D");
            for (int i = 0; i < 46; ++i)
            {
                Console.Write("-");
            }
            Console.WriteLine();

            for (int i = 0; i < studentScores.GetLength(0); i++)
            {
                Console.Write($"学生{i + 1}\t|\t");
                for (int j = 0; j < studentScores.GetLength(1); j++)
                {
                    Console.Write($"{studentScores[i, j]}\t");
                }

                Console.WriteLine();
            }

        }

        public override void Question04()
        {
            int[,] studentScores =
            {
                { 52, 71, 61, 47 },
                { 6, 84, 81, 20 },
                { 73, 98, 94, 95 }
            };

            float[] average = new float[4];
            int total = 0;

            Console.WriteLine("\t|\t科目A\t科目B\t科目C\t科目D\t|\t合計点");
            for (int i = 0; i < 66; ++i)
            {
                Console.Write("-");
            }
            Console.WriteLine();

            for (int i = 0; i < studentScores.GetLength(0); i++)
            {
                Console.Write($"学生{i + 1}\t|\t");

                for (int j = 0; j < studentScores.GetLength(1); j++)
                {
                    Console.Write($"{studentScores[i, j]}\t");
                    total += studentScores[i, j];
                    average[j] += studentScores[i, j];

                }
                Console.Write($"|\t{total}");
                total = 0;
                Console.WriteLine();
                
            }
            Console.Write("平均点\t|\t");
            foreach (var score in average)
            {
                Console.Write($"{Math.Round(score / 3.0f, 1)}\t");
            }
            Console.Write("|");
        }

        public void Question05()
        {
            try
            {
                int count = 0;
                int[] numbers = new int[100];
                for (int i = 0; i < numbers.Length; i++)
                {
                    Console.Write($"{i + 1}件目の入力：");
                    numbers[i] = int.Parse(Console.ReadLine());
                    if (numbers[i] < 0)
                    {
                        break;
                    }
                    count++;
                }

                Console.WriteLine("----並び替え後----");

                var query = numbers.ToList().Take(count).OrderBy((x) => x);

                foreach (var i in query)
                {
                    Console.WriteLine(i);
                }
            }
            catch (Exception e)
            {
                Console.Error.WriteLine(e.Message);
            }
        }
    }
}
